tests/test_dereference.py

```diff
@@ -125,6 +128,21 @@ class Group(Document):
                 [m for m in group_obj.members]
                 self.assertEqual(q, 2)
 
+        # Delete one User so one of the references in the
```

Comment:
Could you separate this into its own test method? This test method is already quite long and I believe that it would emphasize this particular test-case better. You may also want to check the `._dereferenced` correctness in this test

Reply:
Good point. I've fixed this in https://github.com/MongoEngine/mongoengine/pull/2007/commits/fdcaca42ae6145ac102ec3475bdc8c550a7a604a